components/Profile/SettingsPopup.jsx

```diff
@@ -1,54 +1,135 @@
+import React, { useState } from 'react';
 import {
-  Menu, MenuOptions, MenuOption, MenuTrigger,
-} from 'react-native-popup-menu';
-import Proptypes from 'prop-types';
+  Alert, Modal, StyleSheet, Pressable, View, TouchableOpacity, Image,
+} from 'react-native';
 import styled from 'styled-components/native';
 import { Ionicons } from '@expo/vector-icons';
+import firebase from 'firebase';
+import { useNavigation } from '@react-navigation/native';
 
-const StyledText = styled.Text`
-   font-family: 'Montserrat_500Medium';
-   font-size: 12px;
+const Title = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  font-size: 24px;
 `;
 
-const SettingsPopup = (props) => {
-  const { style, options, triggerSize } = props;
+const SubTitle = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  font-size: 20px;
+`;
+
+const Buttontext = styled.Text`
+  font-family: 'Montserrat_500Medium';
+  font-size: 16px;
+`;
+const styles = StyleSheet.create({
+  centeredView: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: 10,
+  },
+  modalView: {
+    height: 500,
+    width: 350,
+    margin: 10,
+    backgroundColor: 'white',
+    borderRadius: 20,
+    padding: 15,
+    alignItems: 'center',
+    shadowColor: '#000',
+    shadowOffset: {
+      width: 0,
+      height: 2,
+    },
+    shadowOpacity: 0.25,
+    shadowRadius: 4,
+    elevation: 5,
+  },
+  button: {
+    backgroundColor: '#E2E2E2',
+    borderRadius: 20,
+    padding: 5,
+    paddingLeft: 15,
+    paddingRight: 15,
+    margin: 10,
+  },
+  colorButton: {
+    backgroundColor: 'black',
+    padding: 1,
+    borderColor: 'black',
+    margin: 5,
+  },
+});
 
+const SettingsPopup = (props) => {
+  const { triggerSize } = props;
+  const [modalVisible, setModalVisible] = useState(false);
+  const navigation = useNavigation();
   return (
-    <Menu style={style}>
-      <MenuTrigger>
-        <Ionicons name="ios-settings" size={triggerSize} color="black" />
-      </MenuTrigger>
-      <MenuOptions customStyles={{
-        optionsContainer: {
-          backgroundColor: '#EDEDF0', borderRadius: 10, padding: 5, width: 'auto',
-        },
-      }}
+    <View>
+      <Modal
+        animationType="slide"
+        transparent
+        visible={modalVisible}
+        onRequestClose={() => {
+          Alert.alert('Modal has been closed.');
+          setModalVisible(!modalVisible);
+        }}
       >
-        {options.map((option, index) => (
-          // eslint-disable-next-line react/no-array-index-key
-          <MenuOption key={option.text + index} onSelect={option.onPress}>
-            <Text>
-              <StyledText>{option.text}</StyledText>
-            </Text>
-          </MenuOption>
-        ))}
-
-      </MenuOptions>
-    </Menu>
+
+        <TouchableOpacity style={styles.centeredView} onPress={() => setModalVisible(!modalVisible)}>
+          <TouchableOpacity style={styles.modalView} onPress={() => setModalVisible(modalVisible)}>
```

Comment:
Why are these touchableOpacities? Do Views not work for these and why do you need the onPress stuff?

Reply:
I had to use touchable opacity and on presses cuz its like the only workaround for making modal close when u touch out of it